testing/mock_test.ts

```diff
@@ -1761,12 +1782,12 @@ Deno.test("returnsThis", () => {
 
 Deno.test("returnsArg", () => {
   let callback = returnsArg(0);
-  assertEquals(callback(), undefined);
+  assertThrows(() => callback());
   assertEquals(callback("a"), "a");
   assertEquals(callback("b", "c"), "b");
   callback = returnsArg(1);
-  assertEquals(callback(), undefined);
-  assertEquals(callback("a"), undefined);
+  assertThrows(() => callback());
+  assertThrows(() => callback("a"));
```

Comment:
Following on from this conversation https://github.com/denoland/deno_std/pull/4084#discussion_r1441182192 the choice was either changing the type or changing the runtime behaviour, you settled on changing the runtime behaviour so the tests needed to change to account for the new behaviour.

Reply:
Oh! I see. Hm... Ok. I now better understand how it affected the public API. On second thought, it might be better to add `undefined` to the return type instead of error-checking. Sorry, are you able to do that?

Reply:
No problem, I could have made it more clear. I'll push the change soon.